scripts/globals/chocobo_riding_game.lua

```diff
@@ -0,0 +1,209 @@
+-----------------------------------
+xi = xi or {}
+xi.chocoboGame = xi.chocoboGame or {}
+
+local raceData =
+{
+    [xi.zone.BASTOK_MINES] =
+    {
+        [xi.zone.EAST_SARUTABARUTA] = { reward = xi.item.MIRATETES_MEMOIRS,  raceTimes = { 1950, 2040 }, npc = bastokID.npc.AZETTE,  eventParam = 2, finishEvent = 901 },
+        [xi.zone.WEST_RONFAURE]     = { reward = xi.item.DRAGON_CHRONICLES,  raceTimes = { 1214, 1260 }, npc = bastokID.npc.QUELLE,  eventParam = 0, finishEvent = 55  },
+        [xi.zone.ROLANBERRY_FIELDS] = { reward = xi.item.BASTOK_MINES_GLYPH, raceTimes = { 1130, 1200 }, npc = bastokID.npc.EULAPHE, eventParam = 3, finishEvent = 901 },
+    },
+    [xi.zone.WINDURST_WOODS] =
+    {
+        [xi.zone.SOUTH_GUSTABERG]      = { reward = xi.item.MIRATETES_MEMOIRS,    raceTimes = { 1950, 2084 }, npc = windurstID.npc.AMIMI,   eventParam = 1, finishEvent = 907 },
+        [xi.zone.WEST_RONFAURE]        = { reward = xi.item.MIRATETES_MEMOIRS,    raceTimes = { 1717, 1800 }, npc = windurstID.npc.SARIALE, eventParam = 0, finishEvent = 55  },
+        [xi.zone.SAUROMUGUE_CHAMPAIGN] = { reward = xi.item.WINDURST_WOODS_GLYPH, raceTimes = {  930,  970 }, npc = windurstID.npc.ORLAINE, eventParam = 3, finishEvent = 901 },
+    },
+    [xi.zone.SOUTHERN_SAN_DORIA] =
+    {
+        [xi.zone.SOUTH_GUSTABERG]   = { reward = xi.item.DRAGON_CHRONICLES,   raceTimes = { 1200, 1248 }, npc = sandoriaID.npc.CAMEREINE, eventParam = 1, finishEvent = 907 },
+        [xi.zone.EAST_SARUTABARUTA] = { reward = xi.item.MIRATETES_MEMOIRS,   raceTimes = { 1699, 1800 }, npc = sandoriaID.npc.EMOUSSINE, eventParam = 2, finishEvent = 901 },
+        [xi.zone.BATALLIA_DOWNS]    = { reward = xi.item.EAST_SANDORIA_GLYPH, raceTimes = {  795,  900 }, npc = sandoriaID.npc.MEUNEILLE, eventParam = 3, finishEvent = 906 },
+    },
+    [xi.zone.KAZHAM] =
+    {
+        [xi.zone.YUHTUNGA_JUNGLE] = { reward = xi.item.MIRATETES_MEMOIRS, raceTimes = { 150, 240 }, npc = kazhamID.npc.TIELLEQUE, eventParam = 3, finishEvent = 208 },
+    },
+}
+
+-- Rewards that all races give
+local defaultRewards =
+{
+    xi.item.CHOCOBO_TICKET,
+    xi.item.BUNCH_OF_GYSAHL_GREENS
+}
+
+-- Checks if the NPC can start a race and the player hasn't already participated this week
+xi.chocoboGame.raceCheck = function(player, npc, zone)
+    local zoneId = zone:getID()
+
+    -- Early false return if zone isn't a racing zone
+    if not raceData[zoneId] then
+        return false
+    end
+
+    -- TODO: Add time available check
+    for destination, data in pairs(raceData[zoneId]) do
+        if
+            npc:getID() == data.npc and
+            player:getCharVar('[ChocoGame]NextEntryTime') == 0
+            -- and correct race time
+        then
+            return destination
+        end
+    end
+
+    return false
+end
+
+-- Get the event param to tell the player which city race they are offering
+xi.chocoboGame.renterOnTrigger = function(player, destination, eventSucceed)
```

Comment:
With this and onEventFinish, I'd almost prefer they're named something different if non-conforming to standard functions (onTrigger usually implies player, npc as args, whereas onEventFinish is player, csid, option, npc).

Reply:
Renaming them to startRaceEvent and beginRace